apps/flutter_exercise/lib/views/widgets/login_widgets/form.dart

```diff
@@ -0,0 +1,231 @@
+              child: const Row(
+                children: [
+                  Text(
+                    "¿Ya tienes cuenta?",
+                    style: TextStyle(
+                      color: Color.fromARGB(255, 23, 57, 25),
+                    ),
+                  ),
+                  Spacer(),
+                  Text(
+                    "Inicia sesión", //Por qué no pilla el paddingleft??
+                    style: TextStyle(
+                      decoration: TextDecoration.underline,
+                      color: Color.fromARGB(255, 23, 57, 25),
+                    ),
+                  )
+                ],
+              ),
+            )
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+// Funcion que devuelve widget etiqueta de texto con el nombre del campo
+Widget _textLabel(String tipo) {
+  return Row(children: [
+    Padding(
+      padding: const EdgeInsets.only(left: 8.0),
+      child: Text(tipo[0].toUpperCase()+tipo.substring(1),
+          style: const TextStyle(
+              color: Color.fromARGB(255, 23, 57, 25),
+              fontWeight: FontWeight.bold)),
+    ),
+    const Text(
+      "*",
+      style: TextStyle(
+          color: Color.fromARGB(255, 255, 0, 0), fontWeight: FontWeight.bold),
+    )
+  ]);
+}
+
+// Funcion que devuelve widget campo de formulario
+Widget _textForm(String tipo, String pLabelText) {
+  return Padding(
+    padding: const EdgeInsets.all(8.0),
+    child: Material(
+      elevation: 5.0,
+      borderRadius: BorderRadius.circular(10.0),
+      shadowColor: const Color.fromARGB(255, 91, 118, 92),
+      child: TextFormField(
+        decoration: InputDecoration(
+            fillColor: const Color.fromARGB(255, 247, 255, 247),
+            filled: true,
+            contentPadding: const EdgeInsets.all(10.0),
+            prefixIcon: _getIcon(tipo),
+            labelText: "$pLabelText $tipo",
+            border: OutlineInputBorder(
+                borderRadius: BorderRadius.circular(10.0),
+                borderSide: BorderSide.none)),
+        // The validator receives the text that the user has entered.
+        validator: (value) {
+          if (value == null || value.isEmpty) {
+            return '$pLabelText $tipo';
+          }
+          return null;
+        },
+      ),
+    ),
+  );
+}
+
+// Función para obtener prefixIcon de _textForm
+Icon _getIcon(String tipo) {
+  switch (tipo) {
+    case 'nombre':
+      return const Icon(Icons.account_circle_outlined,
+          color: Color.fromARGB(255, 91, 118, 92));
+    case 'email':
+      return const Icon(Icons.email_outlined,
+          color: Color.fromARGB(255, 91, 118, 92));
+    case 'contraseña':
+      return const Icon(Icons.key, color: Color.fromARGB(255, 91, 118, 92));
+    default:
+      return const Icon(null);
+  }
+}
+
+// Funcion que devuelve Column con _textLabel y _textForm
+Widget _fieldForm(String tipo, [String pLabelText = "Introduce tu "]) {
```

Comment:
@Mazit0 You should add a boolean parameter if field is required or not

Reply:
Esto no lo entiendo o me pilla sin neuronas... me podrías pasar ejemplo?